### PR TITLE
Issue 13 fixed, features added

### PR DESCRIPTION
**Pull request by starrylight90**
I have added the following features **"
Check user is exist or not. If not, return with error message.
Update, User-App, User-Activity, User-Rank information.
Generate access token and refresh token.
Return access token."**

### CODE REVIEW

File: backend/models/user.js [hunk 1]
```diff
@@ -58,6 +58,25 @@ const userSchema = new mongoose.Schema({
     default: false,
   },
 
+  userRank: {
```
Comment (chassitylead) [inline backend/models/user.js:61]: UserRank, UserActivity and UserApp should be the individual model.
You can find the collection structure of three collections from reputation graph.

Comment (starrylight90) [reply backend/models/user.js:61]: Okay, on it

Comment (starrylight90) [reply backend/models/user.js]: I have made the required changes for adding individual model schemas

File: backend/models/user.js [hunk 2]
```diff
@@ -15,6 +15,38 @@ const mongoose = require("mongoose");
 const crypto = require("crypto");
 const uniqueValidator = require("mongoose-unique-validator");
 
+// Define the UserRank schema
+const userRankSchema = new mongoose.Schema({
```
Comment (chassitylead) [inline backend/models/user.js:19]: You'd better create new file userRank.js for userRankSchema definition.
We will have many schemas and we couldn't put all of them in one file.
It is not a good approach for project maintainability.

File: backend/models/user.js [hunk 3]
```diff
@@ -15,6 +15,38 @@ const mongoose = require("mongoose");
...
+  userRankID: { type: String, required: true, unique: true },
+  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
+  rank: { type: mongoose.Schema.Types.ObjectId, ref: 'Rank', required: true },
+  app: { type: mongoose.Schema.Types.ObjectId, ref: 'App', required: true },
+  dateAchieved: { type: Date, required: true },
+  isActive: { type: Boolean, default: false },
+});
+
+// Define the UserActivity schema
+const userActivitySchema = new mongoose.Schema({
```
Comment (chassitylead) [inline backend/models/user.js:29]: You'd better create new file userActivity.js for userActivitySchema definition.
We will have many schemas and we couldn't put all of them in one file.
It is not a good approach for project maintainability.

File: backend/models/user.js [hunk 4]
```diff
@@ -15,6 +15,38 @@ const mongoose = require("mongoose");
...
+  userActivityID: { type: String, required: true, unique: true },
+  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
+  activity: { type: mongoose.Schema.Types.ObjectId, ref: 'Activity', required: true },
+  datePerformed: { type: Date, required: true },
+  pointsEarned: { type: Number, required: true },
+});
+
+// Define the UserApp schema
+const userAppSchema = new mongoose.Schema({
```
Comment (chassitylead) [inline backend/models/user.js:38]: You'd better create new file userApp.js for userAppSchema definition.
We will have many schemas and we couldn't put all of them in one file.
It is not a good approach for project maintainability.

Comment (starrylight90) [reply backend/models/user.js]: I have created 3 new js files for all schema definitions and committed it.

**Pull request merged**